pyrate/core/orbital.py

```diff
@@ -227,15 +227,13 @@ def independent_orbital_correction(ifg_path, params):
     fullres_dm = get_design_matrix(ifg0, degree, intercept=intercept)
 
     ifg = shared.dem_or_ifg(ifg_path) if isinstance(ifg_path, str) else ifg_path
-    ifg_path = ifg.data_path
+    multi_path = MultiplePaths(ifg.data_path, params)
+    orb_on_disc = MultiplePaths.orb_error_path(ifg.data_path, params)
 
-    multi_path = MultiplePaths(ifg_path, params)
-    fullres_ifg = ifg  # keep a backup
-    orb_on_disc = MultiplePaths.orb_error_path(ifg_path, params)
     if not ifg.is_open:
         ifg.open()
-
     shared.nan_and_mm_convert(ifg, params)
+    fullres_ifg = ifg  # keep a backup
```

Comment:
If the idea is to keep a copy of the numpy phase data before the orbital correction, this line won't achieve that - `x = y` for numpy arrays just creates a new pointer to the same data, so any changes to `y` will also show up in `x`. I think this still applies to attributes of the `Ifg` class, but it should be easy to test this in a python prompt by importing the pyrate `shared` module, creating an Ifg and checking its behaviour. I'm not sure what this "backup" is used for, but it seems this problem has been here since before the PR so it's worth checking it isn't breaking anything else.

Reply:
Yes, I believe the intent is to have two separate yet identical versions of the data at this point.